### PR TITLE
CheckBoxRow Component

### PR DESCRIPTION
**Pull request by ratnadeep007**
- Added `TUICheckBoxRow` component
- Added variants of `TUICheckBoxRow` in kitchen sink app
- Updated `README` with How to run kitchen skin app section

Screenshot 2024-01-24 at 2 43 57 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,4 +35,15 @@ Tarka UI Kit is a reusable component library for building Flutter apps, based on
 - TUITag
 - TUIInputField
 - TUISlider
-- TUIRangeSlider
\ No newline at end of file
+- TUIRangeSlider
```
Comment (Ibrahimhass) [inline README.md:38]: Please add the component to the Readme.
`TUICheckboxRow`

File: example/lib/main.dart [hunk 2]
```diff
@@ -2,6 +2,7 @@ import 'package:example/styles/colors.dart';
 import 'package:fluentui_system_icons/fluentui_system_icons.dart';
 import 'package:flutter/material.dart';
 import 'package:tarka_ui/components/button/style.dart';
+import 'package:tarka_ui/components/checkboxrow/checkboxrow.dart';
```
Comment (Ibrahimhass) [inline example/lib/main.dart:5]: maybe we can rename the file name to checkbox_row (following snake casing)

File: example/lib/main.dart [hunk 3]
```diff
@@ -622,6 +623,78 @@ class _HomePageState extends State<HomePage> {
...
+              enableMixedState: true,
+              title: "Title",
+              description: "Description",
+              backgroundDark: true,
+              state: TUICheckBoxRowState.checked,
+              onChanged: (state) {
+                print("The tapped index is $state");
+              },
+            ),
+            const SizedBox(height: 5.0),
```
Comment (Ibrahimhass) [inline example/lib/main.dart:697]: Please use a consistent vertical spacing of 8.0 pixels as we in the other parts of the example app.

File: lib/components/checkboxrow/checkboxrow.dart [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+      child: Container(
+        decoration: getBoxDecorationRow(context),
+        padding: const EdgeInsets.symmetric(vertical: 9.0, horizontal: 10.0),
+        child: child,
+      ),
+    );
+  }
+
+  CrossAxisAlignment getRowCrossAxisAlignment(String description) {
+    if (description.isNotEmpty) return CrossAxisAlignment.start;
```
Comment (Ibrahimhass) [inline lib/components/checkboxrow/checkboxrow.dart:112]: can we use a ternary operator here?

File: lib/components/checkboxrow/checkboxrow.dart [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    return CrossAxisAlignment.center;
+  }
+
+  EdgeInsets getMarginIfDescription(context, String descrption) {
+    if (descrption.isNotEmpty) return const EdgeInsets.only(top: 3.0);
+    return const EdgeInsets.only(top: 0.0);
```
Comment (Ibrahimhass) [inline lib/components/checkboxrow/checkboxrow.dart:118]: please try to use a ternary operator here as well.

File: lib/components/checkboxrow/checkboxrow.dart [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+  }
+
+  BoxDecoration getBoxDecorationRow(context) {
+    final theme = TUITheme.of(context);
+    Color backgroundColor = Colors.transparent;
+    if (widget.backgroundDark) {
+      backgroundColor = theme.colors.background;
+    }
+    return BoxDecoration(
```
Comment (Ibrahimhass) [inline lib/components/checkboxrow/checkboxrow.dart:127]: add newlines before returns and before if checks like on line 124, and other similar occurrences.

File: lib/components/checkbox_row/checkbox_row.dart [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+  EdgeInsets getMarginIfDescription(context, String descrption) {
+    return descrption.isNotEmpty
+        ? const EdgeInsets.only(top: 3.0)
+        : const EdgeInsets.only(top: 0.0);
+  }
+
+  BoxDecoration getBoxDecorationRow(context) {
+    final theme = TUITheme.of(context);
+    Color backgroundColor = Colors.transparent;
+    if (widget.backgroundDark) {
```
Comment (Ibrahimhass) [inline lib/components/checkbox_row/checkbox_row.dart:126]: please add a newline here as well

Comment (ratnadeep007) [reply lib/components/checkbox_row/checkbox_row.dart:127]: My bad, missed that, done

**Review by Ibrahimhass — APPROVED** [review]

**Pull request merged**